src/Register.php

```diff
@@ -67,8 +67,9 @@ public static function register_widgets() {
 	private static function register_widget( $widget_class ) {
 		register_widget( $widget_class );
 
-		if ( method_exists( $widget_class, 'post_registration' ) ) {
-			call_user_func( [ $widget_class, 'post_registration' ] );
+		$callback = [ $widget_class, 'post_registration' ];
+		if ( is_callable( $callback, true ) ) {
```

Comment:
Yes, otherwise please specify a specific situation with a specific code that points that otherwise please refer to the PHP docs.

> On Jun 1, 2016, at 8:14 AM, Adam Fenton <notifications@github.com> wrote:
> 
> In src/Register.php <https://github.com/moxie-lean/wp-widgets/pull/23#discussion_r65357046>:
> 
> > @@ -67,8 +67,9 @@ public static function register_widgets() {
> >  	private static function register_widget( $widget_class ) {
> >  		register_widget( $widget_class );
> >  
> > -		if ( method_exists( $widget_class, 'post_registration' ) ) {
> > -			call_user_func( [ $widget_class, 'post_registration' ] );
> > +		$callback = [ $widget_class, 'post_registration' ];
> > +		if ( is_callable( $callback, true ) ) {
> I had to change an instance of this to method_exists because is_callable didn't work as expected (it failed when the method didn't exist at at all). Have you tested it in all cases - eg the method does exist, the method doesn't exist and the method exists but is private?
> 
> I also read that method_exists is faster than is_callable, though I'm not too concerned about performance in this case.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/moxie-lean/wp-widgets/pull/23/files/da2b2163ee75d6c22433b4221b9e2f763c6e4480#r65357046>, or mute the thread <https://github.com/notifications/unsubscribe/ADvViaXEnFTvjrKETra7afw16-9nB6_iks5qHYW3gaJpZM4IlAs3>.
>